dbms/src/Functions/FunctionsGeo.cpp

```diff
@@ -246,6 +251,181 @@ class FunctionPointInPolygon : public IFunction
 
 };
 
+const size_t GEOHASH_MAX_TEXT_LENGTH = 16;
+
+// geohashEncode(lat float32/64, lon float32/64, length UInt8) => string
+class FunctionGeohashEncode : public IFunction
+{
+public:
+    static constexpr auto name = "geohashEncode";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionGeohashEncode>(); }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 3; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        validateArgumentType(*this, arguments, 0, isFloat, "float");
+        validateArgumentType(*this, arguments, 1, isFloat, "float");
+        validateArgumentType(*this, arguments, 2, isInteger, "integer");
```

Comment:
Maybe it makes sense to make the third parameter `precision` optional?

Reply:
Ok, sounds like a good idea. In fact, current implementation uses `0` as default value precision, meaning 'max value for precision'.

Implementation-wise, does it means that (in case of missing 3rd parameter) I should just create a const column with a single value and use it as if it was provided by the block?

Reply:
You can check number of arguments in the function `executeImpl()`.

Reply:
fixed